spec/features/log_in_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe 'Login page', type: :feature do
+  before(:example) do
+    visit('/users/sign_in')
+  end
+
+  it 'The username and password inputs and the "Submit" button should be seen.' do
+    expect(page).to have_xpath("//input[@name = 'user[email]']")
+    expect(page).to have_xpath("//input[@name = 'user[password]']")
+    expect(page).to have_button('Log in')
+  end
+
+  it 'When I click the submit button without filling in the username and the password, I get a detailed error.' do
+    click_button 'Log in'
+
+    expect(page).to have_content('Invalid Email or password.')
+  end
+
+  it 'When I click the submit button after filling in the username and the ' \
+     'password with incorrect data, I get a detailed error.' do
+    fill_in('user_email', with: 'test0@test.com')
+    fill_in('user_password', with: 'wrong_password')
+    click_button 'Log in'
+
+    expect(page).to have_content('Invalid Email or password.')
+  end
+
+  it 'When I click the submit button after filling in the username and ' \
+     'the password with correct data, I am redirected to the root page.' do
+    fill_in('user_email', with: 'test0@test.com')
+    fill_in('user_password', with: 'password')
+    click_button 'Log in'
+
+    expect(page).to have_current_path(root_path)
+  end
+end
```

Comment:
![Screenshot from 2022-07-02 22-52-26](https://user-images.githubusercontent.com/38635158/177014577-8dbbf2f2-79b1-46af-94e3-5785403c4067.png)

- [ ] Like the last code reviewer said kindly fix the error.



- I  run the these codes before `bundle exec rspec`

```
rails db:create
rails db:schema:load
rails db:seed```

Reply:
can you help me, with 2 things:
first, can you provide me with the error log of the tests
second, can you run  in terminal:
rails c test
User.find(1)
and share the log

Reply:
@JuanLPalacios Answered in the slack